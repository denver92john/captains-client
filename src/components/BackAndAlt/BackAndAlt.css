.back-and-alt {
    max-width: 400px;
    margin: auto;
    padding: 5px 10px 20px;
    display: flex;
    justify-content: space-between;
}

.fa-backbutton-icon,
.fa-altbutton-icon {
    font-size: 25px;
}

.backbutton-text,
.altbutton-text {
    display: inline-block;
}

.backbutton-text {
    padding-left: 10px;
}

.altbutton-text {
    padding-right: 10px;
}