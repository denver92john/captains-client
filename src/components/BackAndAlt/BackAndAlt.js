import React from 'react';
import BackButton from '../BackButton/BackButton';
import AltButton from '../AltButton/AltButton';
import './BackAndAlt.css';

function BackAndAlt(props) {
    const {backTo, backWhat, altTo, altWhat} = props;
    return (
        <div className="back-and-alt">
            <BackButton backTo={backTo} backWhat={backWhat} />
            <AltButton altTo={altTo} altWhat={altWhat} />
        </div>
    );
}

export default BackAndAlt;