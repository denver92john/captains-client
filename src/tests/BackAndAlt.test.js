import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import BackAndAlt from '../components/BackAndAlt/BackAndAlt';

describe('BackAndAlt component', () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <BackAndAlt 
                    backTo="/dashboard"
                    backWhat="Dashboard"
                    altTo="/teams"
                    altWhat="Pick List"
                    isList={true}
                />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <BackAndAlt 
                        backTo="/dashboard"
                        backWhat="Dashboard"
                        altTo="/teams"
                        altWhat="Pick List"
                        isList={true}
                    />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})