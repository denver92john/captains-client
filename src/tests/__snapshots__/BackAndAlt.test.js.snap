// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackAndAlt component renders UI as expected 1`] = `
<div
  className="back-and-alt"
>
  <div
    className="backbutton-div"
  >
    <a
      href="/dashboard"
      onClick={[Function]}
    >
      <p
        className="backbutton-text"
      >
        Dashboard
      </p>
    </a>
  </div>
  <div>
    <a
      href="/teams"
      onClick={[Function]}
    >
      <p
        className="altbutton-text"
      >
        Pick List
      </p>
    </a>
  </div>
</div>
`;
